telemetry-core/src/main/java/com/newrelic/telemetry/util/IngestWarnings.java

```diff
@@ -0,0 +1,91 @@
+package com.newrelic.telemetry.util;
+
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class IngestWarnings {
+  private static final Logger logger = LoggerFactory.getLogger(IngestWarnings.class);
+  private static final int MAX_NUMBER_OF_EVENT_ATTRIBUTES =
+      254; // Public documentation says 255, but true maximum is 254
+  private static final int MAX_NUMBER_OF_METRIC_ATTRIBUTES = 100;
+  private static final int MAX_NUMBER_OF_LOG_ATTRIBUTES =
+      254; // Public documentation says 255, but true maximum is 254
+
+  private static final int MAX_ATTRIBUTE_NAME_LENGTH = 255;
+  private static final int MAX_ATTRIBUTE_VALUE_LENGTH = 4096;
+
+  public void eventWarningNumAttributes() {
+    logger.warn(
+        "The number of attributes in this event is greater than the maximum allowed attributes per event.");
+  }
+
+  public void metricWarningNumAttributes() {
+    logger.warn(
+        "The number of attributes in this metric is greater than the maximum allowed attributes per metric.");
+  }
+
+  public void logWarningNumAttributes() {
+    logger.warn(
+        "The number of attributes in this log is greater than the maximum allowed attributes per log.");
+  }
+
+  public void attributeNameWarning(String attributeName) {
+    logger.warn(
+        "The length of the attribute named "
+            + attributeName
+            + " is greater than the maximum length allowed for an attribute name.");
+  }
+
+  public void attributeValueWarning(String attributeValue) {
+    logger.warn(
+        "The value of the attribute "
+            + attributeValue
+            + " is greater than the maximum length allowed for an attribute value.");
+  }
+
+  public void raiseIngestWarnings(Map<String, Object> attributes, String dataType) {
```

Comment:
Two things:

Could you move this method to the top of this class please?  :) 

Second, I think with a small tweak, we can make this more flexible.  Events, Metrics, and Logs all implement Telemetry. So we can accept that interface type instead and no longer need to hard code a string.  

You can get the actual type once inside the IngestWarnings and do the logic from there. You might consider saving that type information to be used in string formatting in the log messages too.

```
 public void raiseIngestWarnings(Map<String, Object> attributes, Telemetry dataType) {
     String name = dataType.getClass().getSimpleName();
  }
```